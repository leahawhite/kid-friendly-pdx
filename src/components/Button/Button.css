.form-button {
  border: 1px solid #00b8e6;
  background-color: #00b8e6;
  color: white;
  font-weight: bold;
  font-size: 18px;
  padding: 8px;
  -moz-border-radius: 8px;
  -webkit-border-radius: 8px;
  -o-border-radius: 8px 12px;
  -ms-border-radius: 8px;
  border-radius: 8px;
  margin: 10px auto;
  align-self: center;
}