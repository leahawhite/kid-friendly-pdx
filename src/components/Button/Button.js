import React from 'react'
import './Button.css'

export default function Button(props) {
  return (
    <button type={props.btnType} className="form-button">
      <span>{props.btnText}</span>
    </button>
  )
}