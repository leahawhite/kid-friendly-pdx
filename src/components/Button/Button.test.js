import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Button from './Button'

describe.only('Button component', () => {
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders empty without props', () => {
    const wrapper = shallow(<Button />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders type and text given props', () => {
    const wrapper = shallow(<Button btnType="submit" btnText="Submit" />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
