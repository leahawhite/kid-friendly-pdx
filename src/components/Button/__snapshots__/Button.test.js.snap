// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders empty without props 1`] = `
<button
  className="form-button"
>
  <span />
</button>
`;

exports[`Button component renders type and text given props 1`] = `
<button
  className="form-button"
  type="submit"
>
  <span>
    Submit
  </span>
</button>
`;
